TST, BUG: test_rbfinterp.py -- test_interpolation_misfit_1d fails at least sporadically on Windows

@treverhines 

The rbf test failure described here is still causing issues: https://github.com/scipy/scipy/pull/14118#issuecomment-849024743

The most recent failure I see is in this log for Windows Python 3.8 64-bit with 64-bit OpenBLAS integers active: 
https://dev.azure.com/scipy-org/df8ffb44-7275-43bc-90f9-aa5c93ef5c51/_apis/build/builds/12443/logs/201

Do I have permission to temporarily disable this test on the windows platform while you investigate so I can proceed with the SciPy `1.7.0` release candidate process? If so, I will `xfail` it for that platform most likely and we can keep this issue open until a fix is available. If it really is the test itself that is the problem and not the underlying algorithm/implementation, that is less concerning at least for me, but I'm not an expert here.

Reproduced on MacOS as well on latest `master`, actually for a different kernel (`gaussian` on MacOS vs. `inverse_multiquadric` on Windows).

`python runtests.py -t scipy.interpolate.tests.test_rbfinterp::TestRBFInterpolatorNeighbors20::test_interpolation_misfit_1d `

```
__________________________________________________________________________________________________________________________ TestRBFInterpolatorNeighbors20.test_interpolation_misfit_1d[gaussian] ___________________________________________________________________________________________________________________________
scipy/interpolate/tests/test_rbfinterp.py:203: in test_interpolation_misfit_1d
    yitp = self.build(x, y, epsilon=0.6, kernel=kernel)(xitp)
scipy/interpolate/_rbfinterp.py:451: in __call__
    shift, scale, coeffs = _build_and_solve_system(
scipy/interpolate/_rbfinterp.py:127: in _build_and_solve_system
    raise LinAlgError(msg)
E   numpy.linalg.LinAlgError: Singular matrix.
```

--- a/.gitignore
+++ b/.gitignore
@@ -147,6 +147,7 @@ scipy/integrate/vodemodule.c
 scipy/integrate/_dop-f2pywrappers.f
 scipy/integrate/lsoda-f2pywrappers.f
 scipy/integrate/vode-f2pywrappers.f
+scipy/interpolate/_rbfinterp_pythran.cpp
 scipy/interpolate/_ppoly.c
 scipy/interpolate/_bspl.c
 scipy/interpolate/interpnd.c

--- a/scipy/interpolate/tests/test_rbfinterp.py
+++ b/scipy/interpolate/tests/test_rbfinterp.py
@@ -200,7 +200,7 @@ class _TestRBFInterpolator:
 
         y = _1d_test_function(x)
         ytrue = _1d_test_function(xitp)
-        yitp = self.build(x, y, epsilon=0.6, kernel=kernel)(xitp)
+        yitp = self.build(x, y, epsilon=5.0, kernel=kernel)(xitp)
 
         mse = np.mean((yitp - ytrue)**2)
         assert mse < 1.0e-4

